### PR TITLE
ascanrulesBeta: added status code check for Anti-Token CSRF Scan Rule

### PR DESCRIPTION
**Pull request by aryangupta701**
- added status code check for base msg and the new msg

### CODE REVIEW

**Comment by thc202** [comment]
The title should be more specific, there are so many add-ons and scan rules that the change could apply to.

Needs tests, help should be updated. Would be good to see the FPs in the tests.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java [hunk 1]
```diff
@@ -123,7 +123,8 @@ public void init() {
     @Override
     public void scan() {
         if ((AlertThreshold.HIGH.equals(getAlertThreshold()) && !getBaseMsg().isInScope())
-                || !getBaseMsg().getResponseHeader().isHtml()) {
+                || !getBaseMsg().getResponseHeader().isHtml()
+                || isPage404(getBaseMsg())) {
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java:127]: So your assumption is a 404 response can never be subject to CSRF?

Comment (aryangupta701) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java:127]: Yes. Isn't is true for majority cases?

Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java:127]: Largely, I didn't mean it was a problem I just wanted to be on the same page.

Comment (aryangupta701) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java:127]: Yes we are on the same page.

**Comment by aryangupta701** [comment]
working on tests now

**Pull request closed without merging**